components/mediation/tasks/org.wso2.micro.integrator.ntask.core/src/main/java/org/wso2/micro/integrator/ntask/coordination/task/db/connector/CoordinatedTaskDBConnector.java

```diff
@@ -0,0 +1,362 @@
+        }
+        debugLogs(taskList);
+        return taskList;
+    }
+
+    private void debugLogs(List<String> tasks) {
+        if (LOG.isDebugEnabled()) {
+            debugLogs(tasks, "Following list of tasks were retrieved from db.");
+        }
+    }
+
+    private void debugLogs(List<String> tasks, String msg) {
+        if (LOG.isDebugEnabled()) {
+            LOG.debug(msg);
+            for (String task : tasks) {
+                LOG.debug(" Task Name [" + task + "].");
+            }
+        }
+    }
+
+    private void printDebugLogs(List<CoordinatedTask> tasks) {
+        if (LOG.isDebugEnabled()) {
+            printDebugLogs(tasks, "Following list of tasks were retrieved from db.");
+        }
+    }
+
+    private void printDebugLogs(List<CoordinatedTask> tasks, String msg) {
+        if (LOG.isDebugEnabled()) {
+            LOG.debug(msg);
+            for (CoordinatedTask task : tasks) {
+                LOG.debug(" Task Name [" + task.getTaskName() + "]  destined node Id [" + task.getDestinedNodeId()
+                                  + "] state [" + task.getTaskState() + "].");
+            }
+        }
+    }
+
+    /**
+     * Removes all the tasks assigned to the node.
+     *
+     * @param nodeId - The node id.
+     */
+    public void removeTasksOfNode(String nodeId) throws SQLException {
+
+        try (Connection connection = getConnection(); PreparedStatement preparedStatement = connection.prepareStatement(
+                TaskQueryHelper.REMOVE_TASKS_OF_NODE)) {
+            preparedStatement.setString(1, nodeId);
+            preparedStatement.executeUpdate();
+        }
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Removed all the tasks of node [" + nodeId + "].");
+        }
+    }
+
+    /**
+     * Remove the task entry from the db.
+     *
+     * @param coordinatedTasks - List of tasks to be removed.
+     */
+    public void removeTasksFromDB(List<String> coordinatedTasks) throws SQLException {
+
+        if (coordinatedTasks.isEmpty()) {
+            return;
+        }
+        try (Connection connection = getConnection(); PreparedStatement preparedStatement = connection.prepareStatement(
+                TaskQueryHelper.DELETE_TASK)) {
+            for (String task : coordinatedTasks) {
+                preparedStatement.setString(1, task);
+                preparedStatement.addBatch();
+            }
+            preparedStatement.executeBatch();
+        }
+        debugLogs(coordinatedTasks, "Following list of tasks were deleted from data base.");
+    }
+
+    /**
+     * Helper method to populate coordinated tasks to the list.
+     *
+     * @param preparedStatement - Statement to be executed to retrieve the list of tasks.
+     * @param taskList          - The list to be populated.
+     * @throws SQLException - Exception.
+     */
+    private void populateCoordinatedTaskNames(PreparedStatement preparedStatement, List<String> taskList)
```

Comment:
I wouldn't like mutating the task list that was passed. Rather creating a new object and returning from here.